templates/ModelDecoderEncoderSnippet.stencil

```diff
@@ -0,0 +1,31 @@
+{% if model.hasDateProperty %}
```

Comment:
Truthfully it is probably okay to just always generate this. I suspect there will be other things than just dates that will require some custom encode/decode logic.